mleap-runtime/src/main/scala/ml/combust/mleap/runtime/transformer/Pipeline.scala

```diff
@@ -47,7 +47,7 @@ case class Pipeline(override val uid: String = Transformer.uniqueName("pipeline"
           oacc ++ tform.outputSchema.fields.map(f => f.name -> f.dataType))
     }
 
-    val actualInputs = (inputs -- outputs.keys).map {
+    val actualInputs = inputs.map {
```

Comment:
What is the problem with getting rid of outputs? I would think that is the right thing to do?

Reply:
I was thinking of a pipeline that looks like this (may not be too realistic in Spark, but the scikit pipelines look a lot like this. And what I wanted to get out of the outputSchema is the actual output of the root pipeline, i.e. just the 'scaled_features' field. This is not actually possible at the moment with the current implementation of inputSchema/outputSchema of the Pipeline.

``` val pipeline: Transformer = new Pipeline().setStages(Array(
    new Pipeline().setStages(Array(
    new VectorAssembler().
    setInputCols(Array("dti", "loan_amount")).
    setOutputCol("features"))),
      new Pipeline().setStages(Array(
    new StandardScaler().
      setInputCol("features").
      setOutputCol("scaled_features"))))) ```

Reply:
Here is the way I think of input schema and output schema:

Input schema: All of the fields that must be in the leap frame for the transformer to execute
Output schema: All of the fields that are added to the leap frame after the transformer executes

Maybe we should actually have 3 schemas though:
1. Input schema: All of the fields that must be in the leap frame for the transformer to execute
2. Intermediate schema: All of the fields that are both inputs/outputs to some stage of the transformer
3. Output schema: All of the fields that are added to the leap frame that are not intermediate fields

Reply:
ok, that makes sense to me. is it ok if I make those changes? we only really need three schemas for the pipeline right?

Reply:
Go ahead and make the changes, yeah I think we only need 3 for pipeline. We may also want to add a fourth:

4. Strict output schema: outputSchema -- intermediateSchema

Reply:
Sounds good.